apps/emqx_rule_engine/src/emqx_rule_events.erl

```diff
@@ -121,8 +128,12 @@ unload() ->
     ).
 
 unload(Topic) ->
```

Comment:
unload should be expanded before calling.
unload_hooks_for_rule/1 should expand topic to matching events,
then for each event, check if there are other rules matching this event, and unhook this specific event if no other rule is selecting it.

Reply:
not sure I understand this comment.  :thinking: 

> unload_hooks_for_rule/1 should expand topic to matching events,
then for each event, check if there are other rules matching this event, and unhook this specific event if no other rule is selecting it.

isn't it doing this already?

Reply:
the caller of unload checks if the topic (e.g. `$event/#`) has any overlapping with other rules, if yes, skip unloading ALL event hooks.
It should instead: enumerate each matching event, check if this is the last rule, unhook if yes.

Reply:
I see now, good catch.  :see_no_evil: